# HG changeset patch
# User Mike Hommey <mh+mozilla@glandium.org>
# Date 1721772453 0
# Node ID 132ddf57eccf452c05dd0704451c148eaf184152
# Parent  9c8ca4c9910b67be130226503e01f264323d2966
Bug 1909380 - Don't set NSPR_LIBS multiple times with --with-system-nspr. r=firefox-build-system-reviewers,ahochheiden

NSPR_LIBS is set via pkg_check_modules("NSPR"...) in the
--with-system-nspr case, which conflicts with the other way we're
setting it.

Differential Revision: https://phabricator.services.mozilla.com/D217354

diff --git a/build/moz.configure/nspr.configure b/build/moz.configure/nspr.configure
--- a/build/moz.configure/nspr.configure
+++ b/build/moz.configure/nspr.configure
@@ -112,19 +112,14 @@ def nspr_config(build_env, c_compiler, f
         lib_dir=lib_dir,
     )
 
 
 # Avoid using obsolete NSPR features
 set_define("NO_NSPR_10_SUPPORT", True)
 
 set_config("NSPR_CFLAGS", nspr_config.cflags, when=nspr_config)
-
-
-nspr_libs = nspr_config.libs | nspr_pkg.libs | dependable([])
-
-
-set_config("NSPR_LIBS", nspr_libs)
+set_config("NSPR_LIBS", nspr_config.libs, when=nspr_config)
 
 set_config("NSPR_INCLUDE_DIR", nspr_config.include_dir, when=nspr_config)
 set_config("NSPR_LIB_DIR", nspr_config.lib_dir, when=nspr_config)
 set_config("NSPR_INCLUDE_DIR", nspr_pkg.include_dir, when=nspr_pkg)
 set_config("NSPR_LIB_DIR", nspr_pkg.lib_dir, when=nspr_pkg)
diff --git a/js/moz.configure b/js/moz.configure
--- a/js/moz.configure
+++ b/js/moz.configure
@@ -1500,21 +1500,23 @@ def js_library_name(symbolversion, js_st
     else:
         return "mozjs"
 
 
 set_config("JS_LIBRARY_NAME", js_library_name)
 add_old_configure_assignment("JS_LIBRARY_NAME", js_library_name)
 
 
-@depends(nspr_libs, instruments)
-def js_config_libs(nspr_libs, instruments):
+@depends(nspr_config.libs, nspr_pkg.libs, instruments)
+def js_config_libs(nspr_libs, nspr_pkg_libs, instruments):
     libs = []
     if nspr_libs:
         libs += nspr_libs
+    if nspr_pkg_libs:
+        libs += nspr_pkg_libs
     if instruments:
         libs += ["-framework", "CoreFoundation"]
     return libs
 
 
 set_config("JS_CONFIG_LIBS", js_config_libs)
 
 #


